src/dex/balancer-v3/balancer-v3-pool.ts

```diff
@@ -411,6 +419,12 @@ export class BalancerV3EventPool extends StatefulEventSubscriber<PoolStateMap> {
     let outputAmountRaw = 0n;
     for (const i of indices) {
       const step = steps[i];
+
+      // If the pool has a hook we fetch latest hook state for use in maths
+      let hookState = undefined;
+      if ('hookAddress' in step.poolState && step.poolState.hookAddress)
+        hookState = hookStateMap[step.poolState.hookAddress];
```

Comment:
What if `hookStateMap[step.poolState.hookAddress][step.pool]` is empty here? for example when new pool with hooks was registered, but `updateHooksState` wasn't invoked yet, so hookStateMap[hook][pool] should be empty

Reply:
Didn't think of that, thanks! In that case we cannot calculate swap result so I've changed to log and return 0.

Reply:
should we return 0n in case `hookStateMap[step.poolState.hookAddress][step.pool]` is empty?

Reply:
![image](https://github.com/user-attachments/assets/6effd9d1-f287-4c84-b132-af3042fc9655)
?

Reply:
Not all hooks will have this pool specific state so I moved the check to the stableSurge specifically for now.